### ISSUE TITLE
RenameKeywords capitalization is off if space-underscore combinations

### ISSUE THREAD

**Issue by ursa-h** [MarketSquare/robotframework-tidy#537]
RenameKeywords turns

`Keyword _with_Problematic_Space_Underscore_Combination`

into

`Keyword With PPoblematic SSace UUderscore CCmbination`

Reason is that removing underscore-space combination in middle produces two spaces (when removing underscores is enabled).
Result is zip-compared for capital letters to output of string.capwords(), which reduces all whitespace to a single space char.
The capitals will be off by the amount of space-underscore combinations in the string.

Fixed this (in a custom transformer we needed anyway) by a regex to replace also space-underscore with single space before the call to string.capwords() in /robotframework-tidy/robotidy/transformers/RenameKeywords.py

```
    def remove_underscores_and_capitalize(self, value: str):
        if self.remove_underscores:
            value = re.sub(r"(_|\s_)+", " ", value)  # replace one or more _ with one space

```

**Comment by bhirsz** [comment]
Thanks for the report and the analysis. By the looks of it, your regex should work fine but I will add some extra tests to check if there aren't other issues as well.

**Comment by bhirsz** [comment]
Fixed - it will be released in 4.4.0 in the next few days.
